src/js/components/Calendar/__tests__/__snapshots__/Calendar-test.js.snap

```diff
@@ -139,7 +139,6 @@ exports[`Calendar date 1`] = `
   padding: 0;
   text-align: inherit;
   line-height: 0;
-  padding: 12px;
```

Comment:
Something seems wrong here. I would not expect the Buttons in Calendar to have padding removed. I noticed that they don't look right in the storybook for this pull request.

Reply:
I compared this PR's Calendar storybook to the master's and couldn't see any problem. (I used Incognito) will you share more details on which story you've seen an issue with the padding?

Reply:
Previously:

![Screen Shot 2019-03-11 at 12 52 06 PM](https://user-images.githubusercontent.com/11637956/54153361-a58fe000-43fc-11e9-99d6-fdf9c7b584dc.png)

In this PR:

![Screen Shot 2019-03-11 at 12 51 37 PM](https://user-images.githubusercontent.com/11637956/54153391-b5a7bf80-43fc-11e9-9e1c-3528f9a66841.png)

Note that the buttons aren't 48x48 anymore.

Reply:
@ericsoderberghp Yep, so this brings us back to the doc note I added on top.
the Calendar is using Button with `icon` prop, hence it has the `plain={true}` applied automatically (see https://github.com/grommet/grommet/blob/4e309dcd1388662d4e0fe5862b84709749fcaf4f/src/js/components/Button/Button.js#L137). 
This PR is addressing `plain` to not show padding hence the paddings were removed from this Calendar story (from the Buttons).
as I mentioned on the PR notes we can discuss whether this functionality is the desired one. IMO I think plain shouldn't have padding, and I'm not sure why we are having that special logic for icon to have an auto plain (I guess its because we don't want the border?), but since this is the current behavior I'm not sure if changing it at this point is the right decision.